src/lib/asl.c

```diff
@@ -0,0 +1,56 @@
+#include <asl.h>
+#include <pwd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <time.h>
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+#include <xhyve/asl.h>
+
+static aslclient asl = NULL;
+static aslmsg log_msg = NULL;
+
+static unsigned char *buf = NULL;
+static size_t buf_size = 0;
+static size_t buf_capacity = 0;
+
+/* asl is deprecated in favor of os_log starting with macOS 10.12.  */
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+/* Grow buf/buf_capacity. */
+static void buf_grow(void)
+{
+	if (buf_capacity == 0) {
+		buf_capacity = 1024;
+		buf = malloc(buf_capacity);
+	} else {
+		buf_capacity *= 2;
+		buf = realloc(buf, buf_capacity);
```

Comment:
Both `malloc` and `realloc` can fail so you need to check here. Take extra care with `realloc` not to leak the original buffer -- although given there isn't much error handling you can do without a lot of reworking perhaps an `xhyve_abort` would be a reasonable choice and then leaks are somewhat immaterial.

You could also simplify using the fact that `realloc(NULL, N)` is equivalent to `malloc(N)`.

Reply:
I did not check that allocation succeeded, because there are many places where this is not done, starting with that very file (https://github.com/moby/hyperkit/blob/b2a7a6a1611d3e7489331762c27c7c6d4de73043/src/lib/uart_emul.c#L650).

What would you recommend?  `perror`?